src/View/SSTemplateParser.php

```diff
@@ -1886,6 +1888,8 @@ function PresenceCheck_Argument(&$res, $sub)
             $res['php'] .= '((bool)'.$sub['php'].')';
         } else {
             $php = ($sub['ArgumentMode'] == 'default' ? $sub['lookup_php'] : $sub['php']);
+            // TODO: kinda hacky - maybe we need a way to pass state down the parse chain so
+            // Lookup_LastLookupStep and Argument_BareWord can produce hasValue instead of XML_val
```

Comment:
This and other comments in here were already in the `.peg` file and were automatically added here when running the peg file to generate this one. This file has not been manually altered at all.